GitUI/CommandsDialogs/BrowseDialog/DashboardControl/DashboardCategory.cs

```diff
@@ -286,29 +286,25 @@ public void Clear()
 
         private void addToItem_Click(object sender, EventArgs e)
         {
-            var toolStripItem = sender as ToolStripItem;
-
-            if (toolStripItem == null)
-            {
-                return;
-            }
-
             if (_repository == null)
             {
                 return;
             }
 
-            foreach (RepositoryCategory newRepositoryCategory in Repositories.RepositoryCategories)
+            if (sender is ToolStripItem toolStripItem)
```

Comment:
❕  Use the following instead:

```csharp
if (!(sender is ToolStripItem toolStripItem))
{
    return;
}
```

Reply:
I used to bail early on type tests in the way you're suggesting (via `as` then `null` test), but with this new language feature I find the negative pattern harder to read, and even though `toolStripItem` is in scope outside of the block, I prefer to see it used within the block. A matter of taste I guess.

Reply:
I understand that the syntax is somewhat of a preference, but between a return-early style and the new pattern matching syntax I would much rather keep the return-early style.

Reply:
In many other cases I totally agree with you. This one feels different to me -- for now. It's a newer language feature. Anyway, I had a commit on my machine that addressed this very issue in a few other places (unrelated to pattern matching) which I just submitted #4622.

Reply:
this proposal kind of negates already merged #4622